### PR TITLE
Use secp256k1::{SecretKey, PublicKey} instead of own types

### PR DESCRIPTION
**Pull request by stevenroose**
Another attempt to rewrite https://github.com/rust-bitcoin/rust-bitcoin/pull/183

I know some don't like the `_uncompressed` variants all that much, but I think it makes more sense like this than to have them stored in a separate struct.

This comes with https://github.com/rust-bitcoin/rust-bitcoin/pull/220 for missing WIF stuff.

### CODE REVIEW

**Comment by stevenroose** [comment]
I based this off Carl's PR, so I didn't see the new `Address::p2upkh` constructor. I used `p2pkh_uncompressed` instead. If either name is preferred, I can change it back.

Also apologies for adding the commas, that's my editor doing that for me :) I can remove them if they clutter the diff.

File: src/util/address.rs [hunk 1]
```diff
@@ -432,29 +442,21 @@ mod tests {
...
+        let addr = Address::p2pkh_uncompressed(&key, Bitcoin);
         assert_eq!(&addr.to_string(), "1QJVDzdqb1VpbDK7uDeyVXy9mR27CJiyhY");
 
         let key = hex_key!(&"03df154ebfcf29d29cc10d5c2565018bce2d9edbab267c31d2caf44a63056cf99f");
         let addr = Address::p2pkh(&key, Testnet);
         assert_eq!(&addr.to_string(), "mqkhEMH6NCeYjFybv7pvFC22MFeaNT9AQC");
     }
 
-    #[test]
-    fn test_p2pk () {
```
Comment (stevenroose) [inline src/util/address.rs]: See https://github.com/rust-bitcoin/rust-bitcoin/pull/222.

**Comment by tamasblummer** [comment]
It is disrespectful of reviewers time if you introduce lots of diffs because of your editor conf.

**Comment by stevenroose** [comment]
> It is disrespectful of reviewers time if you introduce lots of diffs because of your editor conf.

That's why I apologized beforehand and offered to remove them if they were bothering. I'll remove them.

**Comment by tamasblummer** [comment]
If you know you'd have to apologize then just do not do it.

File: src/util/address.rs [hunk 2]
```diff
@@ -100,7 +98,7 @@ impl Address {
     /// This address type is discouraged as it uses more space but otherwise equivalent to p2pkh
     /// therefore only adds ambiguity
     #[inline]
-    pub fn p2upkh(pk: &PublicKey, network: Network) -> Address {
+    pub fn p2pkh_uncompressed(pk: &PublicKey, network: Network) -> Address {
```
Comment (tamasblummer) [inline src/util/address.rs]: The name was fine as is.

File: src/util/address.rs [hunk 3]
```diff
@@ -155,6 +153,19 @@ impl Address {
         }
     }
 
+    /// Create a pay to script address that embeds a witness pay to (uncompressed) public key
+    pub fn p2shwpkh_uncompressed (pk: &PublicKey, network: Network) -> Address {
```
Comment (tamasblummer) [inline src/util/address.rs]: Use of uncompressed keys was discouraged before SegWit. We should not encourage doing this by adding a method to create scripts for them within SegWit.

File: src/util/address.rs (hunk 2 above)
Comment (stevenroose) [reply src/util/address.rs]: Yeah like I said, this sneaked in when rebasing off master. It's more consistent with the rest of the uncompressed variant methods, but I'm fine with changing it back.

**Comment by dongcarl** [comment]
Closing, see IRC and https://github.com/rust-bitcoin/rust-bitcoin/pull/183

**Pull request closed without merging**